rfcs/datepicker-component.md

```diff
@@ -0,0 +1,148 @@
+# DatePicker Component
+
+Represents control to work and pick date, time value or a range of values
+
+## Usage
+
+### Basic usage
+
+```javascript
+import * as React from 'react';
+import {DatePicker} from 'baseui/datepicker';
+
+export default () => <DatePicker />;
+```
+
+### Advanced usage
+
+```javascript
+import * as React from 'react';
+import {DatePicker} from 'baseui/datepicker';
+
+export default () => {
+  return <DatePicker
+    initialState={{
+      value: [new Date(2018, 1, 15), new Date(2018, 8, 25)]
+    }}
+    isRange={true}
+    dateFormat={'MM/DD/YY'}
+    minDate={new Date(2000, 1, 1)}
+    maxDate={new Date(2020, 1, 1)}
+    monthRange={2}
+    onChange={(event, value)=>{console.log(value)}}
+    overrides={{
+      MonthGridDay: props => <div style={{
+        color: props.$inRange ? 'red' : 'blue',
+        borderRadius: '5px',
+      }}>{props.$value}</div>,
+    }}
+    />
+}
+```
+
+## Exports
+
+* `DatePicker`
+* `StatefulDatePicker`
+* `StatefulDatePickerContainer`
+* `StyledRoot`
+* `StyledDateLabel`
+* `StyledDateInput`
+* `StyledMonthAndYearBar`
+* `StyledMonthSelect`
+* `StyledYearSelect`
+* `StyledMonthGrid`
+* `StyledMonthGridLabel`
+* `StyledMonthGridDay`
+* `StyledQuickSelect`
+
+## `DatePicker` API
+
+* `isRange: boolean` - Optional
+  If set, datepicker is to select 2 dates range, so there are two date input and `value` comes as Array of two dates, min and max. Otherwise only one date is selected. Default is `false`
+* `dateFormat: string` - Optional
+  Date format that is accepted by `Date.parse` function. RFC2822 tools.ietf.org/html/rfc2822#section-3.3 standard. Default is `DD/MM/YYYY`
+* `value: Date | Array<Date>` - Optional
+  Selected `Date`. If `isRange` is set, `value` is an array of 2 values.
```

Comment:
"is an array of 2 Date values: [min, max]"

Reply:
ok, but not min and max. Only those to be picked or set as default values